internal/api/v1/namespace/index.go

```diff
@@ -34,12 +35,16 @@ func (oc Controller) Index(c *gin.Context) apierror.APIErrors {
 	for _, org := range orgList {
 		appNames, err := namespaceApps(ctx, cluster, org.Name)
 		if err != nil {
-			return err
+			return apierror.InternalError(err)
```

Comment:
Shouldn't this also check for a missing namespace ?
I mean, you check for missing namespace after getting the services.
IMHO this check should also be done after getting the apps, above.

Reply:
namespaceApps won't generate an error if a namespace is missing. That's because it never runs `organization.Exists` like the `namespaceServices` does (through `services.List`). It's inconsistent but as far as the fix is concerned, it can't break anymore.

Reply:
Oh, I see now. It uses the `ListAppRefs` and simply gets an empty slice ... Ok then.

Reply:
We could achieve consistency by making the `services.List` a bit more resilient, if it didn't check for the existence of the namespace. It could simply set it in the selector and return an empty list if the namespace is gone. But since this is a hot fix, let's not change more than we have to.